homeworks/oleg.borodko_olegborodko/homework_5/task_3.js

```diff
@@ -0,0 +1,41 @@
+function copyObject(obj) {
+    const newObject = {};
+
+    function fill(object, externalKey, ifObject = false) {
+        Object.entries(object).forEach((entry) => {
+            const key = entry[0];
+            const value = entry[1];
+
+            newObject[key] = value;
+            if (value && value.constructor === Object) {
+                fill(value, key, true);
+            } else if (ifObject) {
+                newObject[externalKey] = {};
+                newObject[externalKey][key] = value;
+            } else {
+                newObject[key] = value;
+            }
+        });
+    }
+
+    fill(obj);
+    return newObject;
+}
```

Comment:
you've got the idea, but please make it look like something readable... single `reduce()` statement is more that enough here